### PR TITLE
GetRefreshing - Repeat when clear

### LINKED ISSUE #68: Repeat When Clear functionality
Update/EDIT:
It would be nice to have a repeat when functionality where anytime clear is called a subscriber will be resubscribed.  The new public api method will be `store.getRefreshing(barcode)` a call to store.clear() drives the refresh

We also need to add clearAll(barcode) which will clear both memory and disk cache.

### PR DESCRIPTION
**Pull request by digitalbuddha**
Closes #68 

@PaulWoitaschek lmk if any feedback.  Marking as experimental for now

### CODE REVIEW

**Comment by PaulWoitaschek** [comment]
I think it should respond to `clearAll` events too.

An easy way would be to have a second PublishSubject and merge its emissions with the `Observable<Barcode> filter>`.

File: store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java [hunk 1]
```diff
@@ -264,6 +308,7 @@ public void clearMemory() {
         inFlightRequests.invalidateAll();
         clearDiskIfNoOp();
         memCache.invalidateAll();
+        refreshSubject.onNext(ALL_BARCODE);
```
Comment (digitalbuddha) [inline store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java]: got away with using a single subject but filtering on a particular Barcode that signals to clear all

Comment (PaulWoitaschek) [inline store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java]: Great! Make it private?

File: store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java [hunk 2]
```diff
@@ -90,6 +96,44 @@ private void init(Fetcher<Raw> fetcher,
...
+    public Observable<Parsed> getRefreshing(@Nonnull final BarCode barCode) {
+        return get(barCode).compose(repeatWhenCacheEvicted(barCode));
+    }
+
+    @Nonnull
+    private Observable.Transformer<Parsed, Parsed> repeatWhenCacheEvicted(@Nonnull final BarCode key) {
+        Observable<BarCode> filter = refreshSubject.filter(new Func1<BarCode, Boolean>() {
+            @Override
+            public Boolean call(BarCode barCode) {
+                return key.equals(barCode) || barCode.equals(ALL_BARCODE);
```
Comment (pavlospt) [inline store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java]: @digitalbuddha For consistency you could say 

```java
barcode.equals(key) || barCode.equals(ALL_BARCODE)
```

**Comment by digitalbuddha** [comment]
@PaulWoitaschek this is the first feature that we are not using internally. Let me know if anything is broken :-) thanks for suggesting it, I'll have a snapshot out tomorrow.

**Pull request merged**